mlir/lib/CAPI/Target/LLVMIR.cpp

```diff
@@ -0,0 +1,38 @@
+//===-- LLVMIR.h - C Interface for MLIR LLVMIR Target ---------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM
+// Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir-c/Target/LLVMIR.h"
+#include "llvm-c/Support.h"
+
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Module.h"
+#include <memory>
+
+#include "mlir/CAPI/IR.h"
+#include "mlir/CAPI/Support.h"
+#include "mlir/CAPI/Wrap.h"
+#include "mlir/Target/LLVMIR/ModuleTranslation.h"
+
+using namespace mlir;
+
+LLVMModuleRef mlirTranslateModuleToLLVMIR(MlirOperation module,
+                                          LLVMContextRef context,
+                                          MlirStringRef llvmModuleName) {
```

Comment:
I'm not sure we should take a "llvmModuleName" here. Instead the translation API should just pick the module name from the IR (Module is a SymbolOp and can be named!), and we could round-trip test this.

Reply:
(the reason why I rather not having the llvmModuleName here is that contrary to the C++ API we're trying our best to keep the C API "stable" as much as possible)

Reply:
Good point, we should update the C++ API as well. It was created before modules had names :)

Reply:
should this be done in the c++ api then? And not in the C api wrapper? I simply don't pass the argument?

Reply:
For now i did just that, if we want to get the llvm module name form the mlir module name i guess it should be done within the c++ api and not the wrapper.

Reply:
Yep. This patch can be merged with the API that doesn't take the module name,

Reply:
> Yep. This patch can be merged with the API that doesn't take the module name,

Okay, should be ready then.